dcos/http.py

```diff
@@ -71,7 +71,8 @@ def request(method,
             request.headers)
 
         with requests.Session() as session:
-            response = session.send(request.prepare(), timeout=timeout)
+            response = session.send(request.prepare(), timeout=timeout,
+                                    proxies=util.get_proxy_dict())
```

Comment:
This means that we need to read the config file and parse it every time we do an HTTP request. Can we get around this? Looking at the [documentation](http://docs.python-requests.org/en/latest/user/advanced/#proxies) it looks like `requests` support reading proxy config from environment variables. Should we just overwrite the environment variable if we find those entries in the `dcos.toml` config?